sky/resources.py

```diff
@@ -375,6 +416,26 @@ def _try_validate_instance_type(self) -> None:
                 f'inferred from the instance_type {self.instance_type!r}.')
             self._cloud = valid_clouds[0]
 
+    def _try_validate_cpu(self) -> None:
+        if self.cpu is None:
+            return
+        if self.instance_type is not None:
+            assert self.cloud is not None
```

Comment:
Because of `_try_validate_instance_type` which is executed before this method. When the instance type is specified by the user, `Resources` infers the cloud provider from the catalogs and set its `self._cloud`.

Reply:
Thanks, please add this as an inline comment to make it clearer for future reading :)

Reply:
Added.